From 8a9a084f8f82fa587eda154956cdc7984f9a9a56 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Mon, 22 Sep 2025 13:23:15 +0200
Subject: [PATCH] Nuke Knox HDM

Keep the One UI 8.5 HDM classes in place, but prevent the enterprise
framework from registering the HDM service and disable secure-key HDM support.
---
 .../enterprise/EnterpriseDeviceManagerServiceImpl.smali       | 4 ++++
 .../server/enterprise/hdm/HdmSakManager.smali                 | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/smali/com/android/server/enterprise/EnterpriseDeviceManagerServiceImpl.smali b/smali/com/android/server/enterprise/EnterpriseDeviceManagerServiceImpl.smali
--- a/smali/com/android/server/enterprise/EnterpriseDeviceManagerServiceImpl.smali
+++ b/smali/com/android/server/enterprise/EnterpriseDeviceManagerServiceImpl.smali
@@ -3471,6 +3471,8 @@

     invoke-static {v0, v3}, Landroid/os/ServiceManager;->addService(Ljava/lang/String;Landroid/os/IBinder;)V

+    goto/16 :goto_unica_hdm_skip
+
     const-string/jumbo v3, "hdm_service"

     new-instance v4, Lcom/android/server/enterprise/hdm/HdmService;
@@ -3735,6 +3737,8 @@

     invoke-static {v4, v3}, Lcom/android/server/enterprise/EnterpriseDeviceManagerServiceImpl$Injector;->addLazySystemService(Lcom/android/server/enterprise/EnterpriseServiceCallback;Ljava/lang/String;)V

+    :goto_unica_hdm_skip
+
     iget-object v0, v1, Lcom/android/server/enterprise/EnterpriseDeviceManagerServiceImpl;->mContext:Landroid/content/Context;

     sget v2, Lcom/android/server/enterprise/appconfig/ApplicationRestrictionsService;->$r8$clinit:I
diff --git a/smali/com/android/server/enterprise/hdm/HdmSakManager.smali b/smali/com/android/server/enterprise/hdm/HdmSakManager.smali
--- a/smali/com/android/server/enterprise/hdm/HdmSakManager.smali
+++ b/smali/com/android/server/enterprise/hdm/HdmSakManager.smali
@@ -770,6 +770,10 @@
 .method public static isSupported(Landroid/content/Context;)Z
     .locals 5

+    const/4 v0, 0x0
+
+    return v0
+
     const-string v0, "HDM - HdmSakManager"

     const-string/jumbo v1, "isSupportSecureKeyService: "
